src/sonic-host-services/scripts/hostcfgd

```diff
@@ -209,29 +369,78 @@ class AaaCfg(object):
+        if 'nas_ip' not in radius_global:
+            nas_ip = self.get_interface_ip("eth0")
+            if len(nas_ip) > 0:
+                radius_global['nas_ip'] = nas_ip
+        if 'nas_id' not in radius_global:
+            nas_id = self.get_hostname()
+            if len(nas_id) > 0:
+                radius_global['nas_id'] = nas_id
+
+        radsrvs_conf = []
+        if self.radius_servers:
+            for addr in self.radius_servers:
+                server = radius_global.copy()
+                server['ip'] = addr
+                server.update(self.radius_servers[addr])
+
+                if 'src_intf' in server:
+                    # RADIUS: Log a message if src_ip is already defined.
+                    if 'src_ip' in server:
+                        syslog.syslog(syslog.LOG_INFO, \
+            "RADIUS_SERVER|{}: src_intf found. May ignore src_ip".format(addr))
+                    # RADIUS: If server.src_intf, then get the corresponding
+                    # src_ip based on the server.ip, and set it.
+                    src_ip = self.get_interface_ip(server['src_intf'], addr)
+                    if len(src_ip) > 0:
+                        server['src_ip'] = src_ip
+
+                radsrvs_conf.append(server)
+            radsrvs_conf = sorted(radsrvs_conf, key=lambda t: int(t['priority']), reverse=True)
+
         template_file = os.path.abspath(PAM_AUTH_CONF_TEMPLATE)
         env = jinja2.Environment(loader=jinja2.FileSystemLoader('/'), trim_blocks=True)
         env.filters['sub'] = sub
         template = env.get_template(template_file)
-        pam_conf = template.render(auth=auth, src_ip=src_ip, servers=servers_conf)
-        with open(PAM_AUTH_CONF, 'w') as f:
+        if 'radius' in auth['login']:
+            pam_conf = template.render(debug=self.debug, trace=self.trace, auth=auth, servers=radsrvs_conf)
+        else:
+            pam_conf = template.render(auth=auth, src_ip=src_ip, servers=servers_conf)
+
+        # Use rename(), which is atomic (on the same fs) to avoid empty file
+        with open(PAM_AUTH_CONF + ".tmp", 'w') as f:
             f.write(pam_conf)
+        os.chmod(PAM_AUTH_CONF + ".tmp", 0o644)
+        os.rename(PAM_AUTH_CONF + ".tmp", PAM_AUTH_CONF)
 
-        # Modify common-auth include file in /etc/pam.d/login and sshd
+        # Modify common-auth include file in /etc/pam.d/login, sshd, and sudo
         if os.path.isfile(PAM_AUTH_CONF):
-            self.modify_single_file('/etc/pam.d/sshd',  [ "'/^@include/s/common-auth$/common-auth-sonic/'" ])
-            self.modify_single_file('/etc/pam.d/login', [ "'/^@include/s/common-auth$/common-auth-sonic/'" ])
+            self.modify_single_file(ETC_PAMD_SSHD,  [ "'/^@include/s/common-auth$/common-auth-sonic/'" ])
+            self.modify_single_file(ETC_PAMD_LOGIN, [ "'/^@include/s/common-auth$/common-auth-sonic/'" ])
+            self.modify_single_file(ETC_PAMD_SUDO, [ "'/^@include/s/common-auth$/common-auth-sonic/'" ])
```

Comment:
sudo does not deal with remote auth. 

Why is this change of behavior?

Reply:
Does forcing the user to re-authenticate when running the sudo depend on site security policy? I believe this is the reason for  adding it to sudo as well for remote authentication. Otherwise, the re-authentication would occur with the local password (which would not exist for remote authenticated users).

Reply:
Again, this is change of behavior. 
Today sudo does not do remote user auth. 
Agree, that some security policy might need. In this case, we need to provide a configurable knob and conditionally enable it for sudo,

Reply:
Took out sudo PAM file modification